test/state/state.cpp

```diff
@@ -181,6 +182,62 @@ std::variant<TransactionReceipt, std::error_code> transition(
         return acc.destructed || (rev >= EVMC_SPURIOUS_DRAGON && acc.erasable && acc.is_empty());
     });
 
-    return TransactionReceipt{gas_used, host.take_logs()};
+    return TransactionReceipt{gas_used, result.status_code == EVMC_SUCCESS, host.take_logs()};
 }
+
+namespace
+{
+inline bytes legacy_tx_to_rlp(uint64_t nonce, const intx::uint256& gas_price, uint64_t gas_limit,
+    const std::optional<evmone::address>& to, const intx::uint256& value, const bytes_view& data,
+    uint8_t v, const intx::uint256& r, const intx::uint256& s)
+{
+    // rlp [nonce, gas_price, gas_limit, to, value, data, v, r, s];
+    return rlp::encode_tuple(nonce, gas_price, gas_limit,
+        to.has_value() ? to.value() : bytes_view(), value, data, v, r, s);
+}
+
+inline bytes eip1559_tx_to_rlp(uint64_t chain_id, uint64_t nonce,
+    const intx::uint256& max_priority_gas_price, const intx::uint256& max_gas_price,
+    uint64_t gas_limit, const std::optional<evmone::address>& to, const intx::uint256& value,
+    const bytes_view& data, const AccessList& /*access_list*/, bool y_parity,
+    const intx::uint256& r, const intx::uint256& s)
+{
+    // TODO: Implement AccessList encoding
+    // tx_type +
+    // rlp [chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit,
+    // to, value, data, access_list, sig_parity, r, s];
+    return bytes({0x02}) +  // Transaction type (eip1559 type == 2)
+           rlp::encode_tuple(chain_id, nonce, max_priority_gas_price, max_gas_price, gas_limit,
+               to.has_value() ? to.value() : bytes_view(), value, data, std::vector<uint8_t>(),
```

Comment:
```suggestion
               to.has_value() ? to.value() : bytes_view(), value, data, bytes_view(),
```

Reply:
It cannot by bytes because bytes is differently encoded in RLP. This has to be empty list `0xc0` not '0x80'

Reply:
Can it be `AccessList{}` ?